pkgs/teos/default.nix

```diff
@@ -0,0 +1,66 @@
+{ lib
+, stdenv
+, rustPlatform
+, fetchFromGitHub
+, llvmPackages
+, openssl
+, perl
+, protobuf
+, rustfmt
+}:
+
+let
+  version = "0.1.1";
+
+  src = fetchFromGitHub {
+    owner = "talaia-labs";
+    repo = "rust-teos";
+    rev = "v${version}";
+    hash = "sha256-6ayn8j2s5lYLIvCcSEo7pB3kLaUJdy3t84Ah+9Q5ezw=";
+  };
+
+  meta = with lib; {
+    # TODO: Use different description for watchtower plugin and tower (teos) pkg
+    description = "A Lightning watchtower compliant with BOLT13, written in Rust";
+    homepage = "https://github.com/talaia-labs/rust-teos";
+    license = licenses.mit;
+    maintainers = with maintainers; [ seberm ];
+    platforms = platforms.unix;
+  };
+
+  cargoPatches = [ ./add-cargo-lock.patch ];
+
+  buildInputs = [ openssl ];
+  nativeBuildInputs = [
+    perl  # used by openssl-sys to configure
+    protobuf
+    rustfmt
+    llvmPackages.clang
+  ];
+
+  LIBCLANG_PATH = "${llvmPackages.libclang.lib}/lib";
+in
+{
+  teos = rustPlatform.buildRustPackage {
+    pname = "teos";
+    cargoSha256 = "sha256-+V7Irf//Lj7VGFY/rdUrSC0hrblnIIYsWgcG4CKLy1w=";
+    buildAndTestSubdir = "teos";
+
+    inherit version src cargoPatches buildInputs nativeBuildInputs LIBCLANG_PATH;
+
+    cargoTestFlags = [
+      "--workspace"
+
+      # Problems with "Too many open files" error
+      "--exclude=teos"
```

Comment:
If I understand this correctly, this means `teos` isn't built at all!
Surely that's not the thing you are looking for, or is it?

Regarding the too many open files issue, I fixed it before by changing the open files soft and hard limits in `/etc/security/limits.conf`:
```
* soft nofile 262144
* hard nofile 524288
```
That was in an ubuntu 18 system.

Reply:
Many Linux distros (including NixOS) have a soft limit of 1024 open files (`NOFILE`).
If you really need to exceed this (which I would doubt in cause of `teos`), your binary should use `setrlimit` to raise the limit to an appropriate value within in the bounds of the hard limit (which is 524288 for most distros).

Reply:
IIRC, this issue happened to me during compilation and not in runtime.

Reply:
Still, users shouldn't have to raise their `NOFILE` soft limit when building teos. Can you open an issue in the teos repo where we can continue discussing this?

Reply:
I have created the issue:
- https://github.com/talaia-labs/rust-teos/issues/119